### PR TITLE
feat: Make search component extendable

### LINKED ISSUE #2209: [feat] Make Search component more extendable
**Does your feature request relate to a specific USWDS component?**
Yes Search

**What USWDS Version is this feature present in?**
Current

**Is your feature request related to a problem? Please describe.**
Many of our props take `inputProps` to pass down to the underlying `<TextInput>` or `<input>` tag. Search should also allow for that, as well as other possible unforseen customization.

**Describe the solution you'd like**
Make it easier to extend or customize the Search/ search related components.

**Describe alternatives you've considered**
- Option one, add `inputProps` to `Search` and pass it down to `TextInput`.
- Option two, make `Search` more composable in general
  - create a `SearchForm` component that encapsulates the search form specific props and use it in the `Search` component
  - create a `SearchButton` component that encapsulates the custom button
  - use these in `Search` and storybook examples
  - We should consider if adding `inputProps` makes sense here or if just an example of building it out separately is enough.

### PR DESCRIPTION
**Pull request by pearl-truss**
# Summary

This PR:
- Adds `inputProps` to `Search` and passes it down to `TextInput`
- Split `Search` into two new components `SearchField` and `SearchButton`

## Related Issues or PRs

closes #2209  

## How To Test

1. Run `yarn storybook` and look at `Components/Search/*` notice the search input and search buttons have been broken out into their own components. And are composed together in `Search`
2. Run `yarn test:coverage` and verify that all tests are passing and that the coverage threshold is being met.

### SearchField
Screen Shot 2022-12-12 at 10 23 54 AM

### SearchButton
Screen Shot 2022-12-12 at 10 24 02 AM

### Search
Screen Shot 2022-12-12 at 10 24 08 AM

### CODE REVIEW

**Review by NamibiaTorres — DISMISSED** [review]
LGTM! Everything is working and passing. Great job Pearl!

File: src/components/Search/SearchField/SearchField.test.tsx
Comment (gidjin) [inline src/components/Search/SearchField/SearchField.test.tsx:5]: question: I wonder if it's possible/worth the time to add a test ensuring that the new `inputProps` are passed down as expected? I can go either way on it so I'll leave it up to you :)

**Review by gidjin — APPROVED** [review]
LGTM! thanks for contributing! If you want your github profile linked in our readme you can run `yarn contributors:add <github user name> code` and commit the changes before you merge.

File: src/components/Search/SearchField/SearchField.test.tsx
Comment (pearl-truss) [reply src/components/Search/SearchField/SearchField.test.tsx:5]: Thanks for the review! I'll add additional test before I merge this

**Review by jcbcapps — DISMISSED** [review]
LGTM! Great work!

**Comment by pearl-truss** [comment]
@gidjin @NamibiaTorres I added the additional test. However, I'm seeing a build issue that I can't reproduce locally, it seems to be a [CircleCI environment](https://app.circleci.com/pipelines/github/trussworks/react-uswds/9388/workflows/040a2470-cdc7-487a-af9c-185354c1e3cb/jobs/9421) issue. I found some context [here](https://discuss.circleci.com/t/git-github-com-permission-denied-publickey-on-repo-that-previously-worked/37763 ). If either of you have seen this issue before please let me know, I'm not sure how to proceed

**Review by gidjin — APPROVED** [review]
LGTM! Thanks for the updates!

**Pull request merged**
